controllers/updateservice_controller.go

```diff
@@ -516,16 +516,36 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
 }
 
 func (r *UpdateServiceReconciler) ensurePolicyEngineRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
```

Comment:
Why not add a new method up front that returns the route, old or new, or nil. Then the rest of this method is fairly unchanged. if nil returned we operate on the new route but if a route is returned this method doesn't care whether it's old or new it just does it's thing (and my comments below are NA).